algorithms/dp/bitmasking_tsp.py

```diff
@@ -1,15 +1,187 @@
+import math
+
+MAX_ROW = 12
+MAX_COLUMN = 12
+MAX_HOUSE = 12
+
+# neighbor nodes position
+D_X = [-1, 0, 0, 1]
+D_Y = [0, 1, -1, 0]
+
+"""
+Check that the given nodes is valid
+
+:param nodes: The nodes to check
+:param nbRow: The number of rows
+:param nbColumn: The number of columns
+:returns: Nothing but raises error if invalid nodes
+"""
+def check_argument(nodes: list, nbRow: int, nbColumn: int):
+    if len(nodes) != nbRow:
+        raise ValueError(f'The number of rows in nodes ({len(nodes)}) does not match the given number of row: {nbRow} !')
+    for row in nodes:
+        if not isinstance(row, list):
+            raise ValueError(f'The given row {row} is not valid !')
+        if len(row) != nbColumn:
+            raise ValueError(f'The number of columns in row {row} does not match the given number of column: {nbColumn} !')
+        for node in row:
+            if node != '.' and node != '*' and node != '#':
+                raise ValueError(f'The given node "{node}" in row {row} is not valid !')
+
```

Comment:
Could you also make sure that _check_argument_ verifies that _nodes_ isn't empty? I ran some tests on this code on a local branch and found that all tests passed except for the test in which nodes=[].

Reply:
Sure, I'll add this special case when checking the argument